### PR TITLE
simpleitk 0.10.0

### PR DESCRIPTION
**Pull request by iMichka**
### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### Updates to existing formula: have you

- [ ] Removed the `revision` line, if any, when bumping the version number?
- [ ] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [x] Not altered the `bottle` section?
- [x] Checked that the tests still pass?

- Bumped version to 0.10.0
- Updated description to comply with brew audit
- Updated position of options to comply with brew audit
- Deprecated examples option to comply with brew audit
- Added option for python 3 support (inspired from the gdcm and vtk formulae)

### CODE REVIEW

**Comment by iMichka** [comment]
Setting up all these python paths, envs and so on is tedious. We are already doing that in gdcm, vtk, and now simpleitk. The insighttoolkit formula could also benefit from it.

It would be great if there was a way to grab that information from homebrew, so that there is a single place where these things get defined. Each time something needs a change, at least the 4 mentioned formulae need to be tweaked.

File: simpleitk.rb [hunk 1]
```diff
@@ -12,54 +12,77 @@ class Simpleitk < Formula
     sha256 "b09618f1801915dd6cdba0c25af6685c5ddbb7dc21c8ad11e178c4e9600ae5f8" => :mavericks
   end
 
+  option "with-examples", "Compile and install various examples"
+  option "with-csharp", "Compile for C sharp"
+  option "with-java", "Compile for Java"
```
Comment (dpo) [inline simpleitk.rb]: This could be removed actually; line 27 takes care of it.

**Comment by dpo** [comment]
It's possible to make things somewhat generic using constructs such as
```ruby
Language::Python.each_python(build) do |python, version|
  system python, *Language::Python.setup_install_args(prefix)
end
```
The formulae over at homebrew/python are instructive. You can use similar constructs in the `test` method.

File: simpleitk.rb (hunk 1 above)
Comment (iMichka) [inline simpleitk.rb]: Right, updated

**Comment by iMichka** [comment]
I will have a look at the more generic construct for later (when I find some time for it). I want to see if I can adapt it for multiple formulae first.

**Comment by iMichka** [comment]
I updated the PR by changing the tar.gz, there is now a SimpleITK-0.10.0-1.tar.gz version that contains the missing file. The maintainers just repackaged the 0.10.0 version. The patch was therefore removed.

**Comment by dpo** [comment]
Are you happy with this?

**Comment by iMichka** [comment]
Yes. The formula was update and all is green. Improving the python setup (also for other formulae) is something I may look at later, if I can find some spare time :)

**Comment by dpo** [comment]
Thank you!

**Comment by dewet22** [comment]
@iMichka After the `-1` patch, brew now misinterprets the version number as just "1". I opened #4820 to address this.

**Pull request closed without merging**